synapse/config/cache.py

```diff
@@ -232,7 +235,18 @@ def read_config(self, config, **kwargs) -> None:
 
         # Backwards compatibility support for the now-removed "expiry_time" config flag.
         expiry_time = cache_config.get("expiry_time")
+
+        if expiry_time and expire_caches:
+            logger.warning(
+                "You have set two incompatible flags, expiry_time and expire_caches. Please only use the "
+                "expire_caches and cache_entry_ttl flags and delete the expiry_time flag as it is "
+                "deprecated."
+            )
         if expiry_time:
+            logger.warning(
+                "Expiry_time is a deprecated flag, please use the expire_caches and cache_entry_ttl flags "
+                "instead."
+            )
```

Comment:
This is very nitty, but I would call these *options* rather than *flags*.
At least in my mind, a flag is a boolean.

Reply:
This is a useful distinction, thank you!